File: Level 1/신규 아이디 추천/solution.js
function solution(new_id) {
  const answer = new_id
    .toLowerCase()
    .replace(/[^a-z0-9-_.]/g, "")
    .replace(/[.]+/g, ".")
    .replace(/^[.]|[.]$/g, "")
    .replace(/^$/g, "a")
    .slice(0, 15)
    .replace(/[.]$/g, "");

  return answer.padEnd(3, answer.at(-1));
}

File: Level 1/신규 아이디 추천/solution.py
import re


def solution(new_id):
    answer = new_id.lower()
    answer = re.sub("[^a-z0-9-_.]", "", answer)
    answer = re.sub("[.]+", ".", answer)
    answer = re.sub("^[.]|[.]$", "", answer)
    answer = answer if answer else "a"
    answer = answer[:15]
    answer = re.sub("[.]$", "", answer)
    return answer if len(answer) > 3 else answer + answer[-1] * (3 - len(answer))
